sdk/communication/Azure.Communication.Common/tests/Pipeline/HMACAuthenticationPolicyTests.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace Azure.Communication.Pipeline
+{
+    public class HMACAuthenticationPolicyTests : PolicyTestBase
+    {
+        private const string secret_key = "68810419818922fb0263dd6ee4b9c56537dbad914aa7324a119fce26778a286e";
+
+        [Test]
+        public async Task TestHMACPolicyProcess()
+        {
+            var shaValue = "47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=";
+            var authPolicy = new HMACAuthenticationPolicy(secret_key);
+            var transport = new MockTransport(new MockResponse(200));
+            await SendGetRequest(transport, authPolicy);
+            var headers = transport.SingleRequest.Headers;
+
+            Assert.True(headers.Contains("Date"));
+            Assert.True(headers.Contains("x-ms-content-sha256"));
+            Assert.True(headers.Contains("Authorization"));
+
+            foreach (HttpHeader h in headers) {
+                if (h.Name == "x-ms-content-sha256") {
+                    Assert.AreEqual(h.Value, shaValue);
```

Comment:
Any reason why not use the format of other examples:
```C#
Assert.True(headers.TryGetValue("x-ms-content-sha256", out string shaValue));
Assert.AreEqual(expectedShaValue, shaValue);
.... same for others
```

Reply:
Oh I see I misunderstood how `TryGetValue` method works. I will definitely make this change